`cargo doc -p hickory-recursor` fails

**Describe the bug**
what the title says

**To Reproduce**
``` console
$ cargo doc -p hickory-recursor
error[E0433]: failed to resolve: could not find `test` in `tokio`
   --> crates/recursor/src/recursor.rs:360:10
    |
360 | #[tokio::test]
    |          ^^^^ could not find `test` in `tokio`

For more information about this error, try `rustc --explain E0433`.
error: could not document `hickory-recursor`
```

**Expected behavior**
The command should succeed

**System:**
 - OS: Ubuntu
 - Architecture: x86_64
 - Version 22.04
 - rustc version: 1.79

**Version:**
Crate: hickory-recursor
Version: 35e2092760b9639a544fcb813754cab59cffe3d5

**Additional context**
Fix incoming.

You may want to consider running something like `cargo ws exec cargo doc` in CI to catch issues like this in the future.

--- a/crates/recursor/src/recursor.rs
+++ b/crates/recursor/src/recursor.rs
@@ -357,6 +357,7 @@ enum RecursorMode {
     },
 }
 
+#[cfg(test)]
 #[tokio::test]
 async fn not_fully_qualified_domain_name_in_query() -> Result<(), Error> {
     use crate::{proto::rr::RecordType, resolver::Name};

